### ISSUE TITLE
TravisCI build is failing

### ISSUE THREAD

**Issue by HarveyHunt** [HarveyHunt/howm#53]
This might be because of changes I have made to the code, or perhaps the required packages have changed.

Can someone on ubuntu please try to compile howm, after installing the packages in [the .travis.yml](https://github.com/HarveyHunt/howm/blob/develop/.travis.yml)

**Comment by HarveyHunt** [comment]
@neo1691 @anshulk

**Comment by neo1691** [comment]
I will give it a look, but I have a tight schedule at office (Where I have my ubuntu box) this week. I should be able to update you next week. I hope someone does this sooner.

**Comment by HarveyHunt** [comment]
Thanks mate, I've been too busy to work on howm recently - just fixing some old bugs now. :+1:

**Comment by loreb** [comment]
On Debian Sid I get the same failure...
The weird thing is that it looks to me like it shouldn't work anywhere!
`xproto.h` includes (very close to the beginning) `xcb.h`, which declares `const xcb_query_extension_reply_t *xcb_get_extension_data(xcb_connection_t *c, xcb_extension_t *ext);`, which uses `xcb_query_extension_reply_t`, which is defined *later* in xproto.h!

TLDR: xcb.h should predeclare the types it's using!

Proof: adding
```
typedef struct xcb_query_extension_reply_t xcb_query_extension_reply_t;
typedef struct xcb_setup_t xcb_setup_t;
```

got scratchpad.c to compile, but then src/op.c failed for the same reason, and so on.

**Comment by loreb** [comment]
Just looked at http://xcb.freedesktop.org/manual/xcb_8h_source.html, it looks like upstream fixed the problem in a different way (`const struct xcb_query_extension_reply_t *xcb_get_extension_data(xcb_connection_t *c, xcb_extension_t *ext)` - notice `struct`) => ubuntu&debian should update their packages.

**Comment by HarveyHunt** [comment]
Thanks for taking a look into this, I have been stupidly busy recently.

We'll just have to sit and wait then. :-(

**Comment by HarveyHunt** [comment]
This has been fixed in 834a1798281eeaa8ed3ca38bc487589a01fbdede by simply including ```xcb.h``` before ```xproto.h```. :-)
